PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/MySQLCompletionContributor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.intellij.runner;
+
+import com.intellij.codeInsight.completion.*;
+import com.intellij.codeInsight.lookup.AutoCompletionPolicy;
+import com.intellij.codeInsight.lookup.LookupElement;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.editor.EditorModificationUtil;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.patterns.PlatformPatterns;
+import com.intellij.util.ProcessingContext;
+import com.microsoft.azure.toolkit.intellij.link.LinkMySQLToModuleDialog;
+import com.microsoft.azure.toolkit.intellij.link.base.LinkType;
+import com.microsoft.azure.toolkit.intellij.link.po.LinkPO;
+import com.microsoft.intellij.AzureLinkStorage;
+import com.microsoft.intellij.helpers.AzureIconLoader;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureIconSymbol;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MySQLCompletionContributor extends CompletionContributor {
+
+    public MySQLCompletionContributor() {
+        extend(CompletionType.BASIC, PlatformPatterns.psiElement(),
+                new CompletionProvider<CompletionParameters>() {
+                    @Override
+                    public void addCompletions(@NotNull CompletionParameters parameters,
+                                               @NotNull ProcessingContext context,
+                                               @NotNull CompletionResultSet resultSet) {
+                        resultSet.addElement(LookupElementBuilder
+                                .create("spring.datasource.url")
+                                .withIcon(AzureIconLoader.loadIcon(AzureIconSymbol.MySQL.BIND_INTO))
+                                .withInsertHandler(new MyInsertHandler())
+                                .withBoldness(true)
+                                .withTypeText("String")
+                                .withTailText(" (Link Azure Datasource for MySQL Service)")
+                                .withAutoCompletionPolicy(AutoCompletionPolicy.SETTINGS_DEPENDENT)
+                        );
+                    }
+                }
+        );
+
+    }
+
+    private class MyInsertHandler implements InsertHandler<LookupElement> {
```

Comment:
make it static if possible.

Reply:
inner private. it should be used only in this file.